forge-game/src/main/java/forge/game/trigger/TriggerLifeLostAll.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Forge: Play Magic: the Gathering.
+ * Copyright (C) 2011  Forge Team
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package forge.game.trigger;
+
+import com.google.common.collect.Maps;
+import forge.game.ability.AbilityKey;
+import forge.game.ability.AbilityUtils;
+import forge.game.card.Card;
+import forge.game.player.Player;
+import forge.game.spellability.SpellAbility;
+import forge.util.Expressions;
+
+import java.util.Map;
+
+/**
+ * <p>
+ * Trigger_LifeLostAll class.
+ * </p>
+ *
+ * 4/27/2023 - this trigger is written for only Ob Nixilis, Captive Kingpin – any future uses will probably need
+ * additional logic
+ *
+ * @author Forge (Northmoc)
+ * @version $Id$
+ */
+public class TriggerLifeLostAll extends Trigger {
+
+    private Map<Player, Integer> triggerMap = Maps.newHashMap();
```

Comment:
Hmn, I think there might be corner cases hiding there where it would get overwritten if it triggers more than once during some other effect

Imo just filter it again, shouldn't really matter for performance